appcues/src/main/java/com/appcues/debugger/screencapture/AndroidViewTargeting.kt

```diff
@@ -0,0 +1,111 @@
+package com.appcues.debugger.screencapture
+
+import android.graphics.Rect
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.children
+import com.appcues.ElementSelector
+import com.appcues.ElementTargeting
+import com.appcues.ViewElement
+import com.appcues.isAppcuesView
+import com.appcues.monitor.AppcuesActivityMonitor
+
+internal data class AndroidViewSelector(
+    var contentDescription: String? = null,
+    var tag: String? = null,
+) : ElementSelector {
+    val isValid: Boolean
+        get() = contentDescription != null || tag != null
+
+    override fun toMap(): Map<String, String> {
+        return mapOf(
+            "contentDescription" to contentDescription,
+            "tag" to tag,
+        ).filterValues { it != null }.mapValues { it.value as String }
+    }
+
+    override fun evaluateMatch(target: ElementSelector): Int {
+        var weight = 0
+
+        (target as? AndroidViewSelector)?.let {
+            if (it.contentDescription != null && it.contentDescription == contentDescription) {
+                weight += 1
+            }
+
+            if (it.tag != null && it.tag == tag) {
+                weight += 1
+            }
+        }
```

Comment:
definitely subject to evolution and change, but this allows for some weighted score based on multiple properties

want to look into using the resource ID as an option as well sometime soon